Content.Server/_Goobstation/Heretic/Abilities/HereticAbilitySystem.Ash.cs

```diff
@@ -20,29 +18,91 @@ public sealed partial class HereticAbilitySystem : EntitySystem
     [Dependency] private readonly TemperatureSystem _temperature = default!;
     public SoundSpecifier JauntExitSound = new SoundPathSpecifier("/Audio/Magic/fireball.ogg");
     public const float RebirthRange = 3f;
+    public TimeSpan BlazingDashDuration = TimeSpan.FromSeconds(5);
 
     private void SubscribeAsh()
     {
         SubscribeLocalEvent<HereticComponent, EventHereticAshenShift>(OnJaunt);
         SubscribeLocalEvent<GhoulComponent, EventHereticAshenShift>(OnJauntGhoul);
         SubscribeLocalEvent<HereticComponent, PolymorphRevertEvent>(OnJauntEnd);
+        SubscribeLocalEvent<HereticComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshSpeed);
 
         SubscribeLocalEvent<HereticComponent, EventHereticVolcanoBlast>(OnVolcano);
+        SubscribeLocalEvent<HereticComponent, EventHereticBlazingDash>(OnBlazingDash);
         SubscribeLocalEvent<HereticComponent, EventHereticNightwatcherRebirth>(OnNWRebirth);
         SubscribeLocalEvent<HereticComponent, EventHereticFlames>(OnFlames);
         SubscribeLocalEvent<HereticComponent, EventHereticCascade>(OnCascade);
     }
 
+    public override void Update(float timeframe)
+    {
+        base.Update(timeframe);
+
+        var dashEndQuery = EntityQueryEnumerator<HereticComponent>();
+        while (dashEndQuery.MoveNext(out var uid, out var hereticComp))
```

Comment:
checking all heretics all the time when only one path actually uses this seems a bit scary. might be worth adding a `BlazingDashComponent` or something to handle this instead? to be fair, i'm not entirely certain on the performance impact that `EntityQueryEnumerator`s have